### ISSUE TITLE
Empty azuracast database

### ISSUE THREAD

**Issue by teardroponfire** [SlvrEagle23/AzuraCast#90]
I'm trying to install AzuraCast on an Ubuntu Server 16.04.1 LTS virtual machine running on my company's internal OpenStack instance.

Here's the install log:

```
Installing AzuraCast (Environment: production)

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [init : debug] ************************************************************
ok: [localhost] => {
    "msg": "Running Ansible on localhost with OS Ubuntu xenial 16.04 x86_64 (production)"
}

TASK [init : debug] ************************************************************
skipping: [localhost]

TASK [init : Update apt] *******************************************************
changed: [localhost]

TASK [init : Install system packages] ******************************************
changed: [localhost] => (item=[u'apt-transport-https', u'curl', u'wget', u'tar', u'build-essential', u'python-software-properties', u'pwgen', u'whois', u'python-pip'])

TASK [init : Update pip components] ********************************************
changed: [localhost] => (item=pip)

TASK [git : Install Git] *******************************************************
ok: [localhost]

TASK [azuracast-user : (Prod) Generate AzuraCast Password] *********************
changed: [localhost]

TASK [azuracast-user : Create Groups] ******************************************
ok: [localhost] => (item=www-data)
changed: [localhost] => (item=admin)

TASK [azuracast-user : Assign User Password] ***********************************
ok: [localhost]

TASK [azuracast-user : Create AzuraCast User] **********************************
changed: [localhost]

TASK [azuracast-user : Modify www-data User] ***********************************
changed: [localhost]

TASK [azuracast-user : (Dev) Modify vagrant User] ******************************
skipping: [localhost]

TASK [azuracast-user : (Dev) Modify www-data User] *****************************
skipping: [localhost]

TASK [azuracast-user : (Dev) Add azuracast User to vagrant Group] **************
skipping: [localhost]

TASK [azuracast-config : Use Default Configuration Files] **********************
changed: [localhost] => (item=db)
changed: [localhost] => (item=influx)
changed: [localhost] => (item=cache)

TASK [azuracast-config : Write current environment to file.] *******************
changed: [localhost]

TASK [azuracast-config : Touch .deploy_run file] *******************************
changed: [localhost]

TASK [azuracast-config : Set Permissions on Base App Folder] *******************
changed: [localhost] => (item=/var/azuracast)

TASK [azuracast-config : Create System Folders] ********************************
changed: [localhost] => (item=/var/azuracast/www_tmp)
changed: [localhost] => (item=/var/azuracast/www_tmp/cache)
changed: [localhost] => (item=/var/azuracast/www_tmp/sessions)
changed: [localhost] => (item=/var/azuracast/www_tmp/proxies)
changed: [localhost] => (item=/var/azuracast/stations)
changed: [localhost] => (item=/var/azuracast/servers)

TASK [azuracast-radio : Install IceCast2 (14.04)] ******************************
skipping: [localhost]

TASK [azuracast-radio : Install IceCast2 (16.04)] ******************************
changed: [localhost]

TASK [azuracast-radio : Link Fallback Error MP3] *******************************
changed: [localhost]

TASK [azuracast-radio : Remove problematic LiquidSoap plugins] *****************
ok: [localhost] => (item=[u'liquidsoap-plugin-all', u'liquidsoap-plugin-opus'])

TASK [azuracast-radio : Install LiquidSoap] ************************************
changed: [localhost] => (item=[u'liquidsoap', u'liquidsoap-plugin-ogg', u'liquidsoap-plugin-vorbis', u'liquidsoap-plugin-lame', u'liquidsoap-plugin-flac', u'liquidsoap-plugin-icecast'])

TASK [azuracast-radio : Download ShoutCast 2 (x86)] ****************************
skipping: [localhost]

TASK [azuracast-radio : Download ShoutCast 2 (x64)] ****************************
changed: [localhost]

TASK [azuracast-radio : Extract ShoutCast 2 Binary] ****************************
changed: [localhost]

TASK [influxdb : Add InfluxDB repo file] ***************************************
changed: [localhost]

TASK [influxdb : Add InfluxDB repo key] ****************************************
changed: [localhost]

TASK [influxdb : Update apt cache] *********************************************
changed: [localhost]

TASK [influxdb : Install python-influxdb] **************************************
changed: [localhost] => (item=influxdb)

TASK [influxdb : Install InfluxDB] *********************************************
changed: [localhost] => (item=[u'influxdb'])

TASK [influxdb : restart influxdb] *********************************************
changed: [localhost]

TASK [influxdb : Pause to allow InfluxDB to spin up] ***************************
Pausing for 15 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [localhost]

TASK [influxdb : Create 'stations' database] ***********************************
changed: [localhost] => (item=stations)

TASK [influxdb : Create 'stations' retention policies] *************************
changed: [localhost] => (item={u'duration': u'5d', u'default': True, u'name': u'15s'})
changed: [localhost] => (item={u'duration': u'2w', u'default': False, u'name': u'1h'})
changed: [localhost] => (item={u'duration': u'INF', u'default': False, u'name': u'1d'})

TASK [influxdb : Manage continuous queries on 'stations'] **********************
changed: [localhost]

TASK [influxdb : Lock down InfluxDB to localhost requests] *********************
changed: [localhost]

TASK [supervisord : Install supervisord] ***************************************
changed: [localhost]

TASK [supervisord : Change supervisord configuration] **************************
changed: [localhost]

TASK [supervisord : Kill legacy radio processes (if upgrading from older version)] ***
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["killall", "-9", "icecast2", "sc_serv", "liquidsoap"], "delta": "0:00:00.004212", "end": "2017-02-02 02:50:18.820065", "failed": true, "rc": 1, "start": "2017-02-02 02:50:18.815853", "stderr": "icecast2: no process found\nsc_serv: no process found\nliquidsoap: no process found", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring

TASK [supervisord : Trigger manual restart of supervisord (if not already running)] ***
changed: [localhost]

TASK [nginx : Install nginx] ***************************************************
changed: [localhost]

TASK [nginx : Create nginx ssl directory] **************************************
changed: [localhost] => (item=/etc/nginx/ssl)

TASK [nginx : Create self-signed SSL cert] *************************************
changed: [localhost]

TASK [nginx : Change default nginx site] ***************************************
changed: [localhost]

TASK [nginx : Turn sendfile off on nginx for local development] ****************
skipping: [localhost]

TASK [php : Add PHP PPA repository] ********************************************
changed: [localhost]

TASK [php : Update apt again] **************************************************
changed: [localhost]

TASK [php : Install PHP Libraries] *********************************************
changed: [localhost] => (item=[u'php7.0-fpm', u'php7.0-cli', u'php7.0-gd', u'php7.0-curl', u'php7.0-xml', u'php7.0-zip', u'php7.0-mysqlnd', u'php7.0-mbstring', u'php7.0-intl'])

TASK [php : List locales] ******************************************************
changed: [localhost]

TASK [php : Create Locales] ****************************************************
changed: [localhost] => (item=de_DE.UTF-8)
changed: [localhost] => (item=es_ES.UTF-8)
changed: [localhost] => (item=fr_FR.UTF-8)
changed: [localhost] => (item=pl_PL.UTF-8)
changed: [localhost] => (item=ru_RU.UTF-8)

TASK [php : Remove Old PHP Libraries] ******************************************
ok: [localhost] => (item=[u'php5-common', u'php5-fpm', u'php5-cli', u'php5-phalcon', u'php7.0-phalcon'])

TASK [php : Configure PHP FPM Pool] ********************************************
changed: [localhost]

TASK [php : Configure php-fpm php.ini] *****************************************
changed: [localhost] => (item={u'option': u'post_max_size', u'value': u'50M'})
changed: [localhost] => (item={u'option': u'short_open_tag', u'value': u'On'})
changed: [localhost] => (item={u'option': u'upload_max_filesize', u'value': u'25M'})

TASK [php : Configure php-cli php.ini] *****************************************
changed: [localhost] => (item={u'option': u'short_open_tag', u'value': u'On'})

TASK [mariadb : (Prod) Generate MySQL Root Password] ***************************
changed: [localhost]

TASK [mariadb : Assign User Password] ******************************************
ok: [localhost]

TASK [mariadb : Add repo file] *************************************************
skipping: [localhost]

TASK [mariadb : Add repo key] **************************************************
skipping: [localhost]

TASK [mariadb : Update apt cache] **********************************************
skipping: [localhost]

TASK [mariadb : Unattended package installation] *******************************
ok: [localhost]

TASK [mariadb : Install MariaDB] ***********************************************
changed: [localhost] => (item=[u'mariadb-server', u'mariadb-client'])

TASK [mariadb : Add configuration] *********************************************
changed: [localhost]

TASK [mariadb : Start and enable service] **************************************
ok: [localhost]

TASK [mariadb : Add empty .my.cnf (if none exists)] ****************************
changed: [localhost]

TASK [mariadb : MySQL Root Account Fix (Ubuntu 16.04)] *************************
changed: [localhost]

TASK [mariadb : Set root Password] *********************************************
changed: [localhost]

TASK [mariadb : Add .my.cnf] ***************************************************
changed: [localhost]

TASK [mariadb : Remove anonymous users] ****************************************
ok: [localhost] => (item=DELETE FROM mysql.user WHERE User='')

TASK [mariadb : Disallow root login remotely] **********************************
ok: [localhost] => (item=DELETE FROM mysql.user WHERE User='root' AND Host NOT IN ('localhost', '127.0.0.1', '::1'))

TASK [mariadb : Remove test database and access to it] *************************
failed: [localhost] (item=DROP DATABASE test) => {"changed": false, "cmd": ["mysql", "--defaults-extra-file=/root/.my.cnf", "-ne", "DROP DATABASE test"], "delta": "0:00:00.005738", "end": "2017-02-02 02:52:53.787422", "failed": true, "item": "DROP DATABASE test", "rc": 1, "start": "2017-02-02 02:52:53.781684", "stderr": "ERROR 1008 (HY000) at line 1: Can't drop database 'test'; database doesn't exist", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring
ok: [localhost] => (item=DELETE FROM mysql.db WHERE Db='test' OR Db='test\_%')

TASK [mariadb : Reload privilege tables] ***************************************
ok: [localhost] => (item=FLUSH PRIVILEGES)

TASK [mariadb : Create MySQL Database] *****************************************
changed: [localhost]

TASK [mariadb : Copy default DB config again] **********************************
ok: [localhost]

TASK [mariadb : Set MySQL Password in Application] *****************************
changed: [localhost]

TASK [composer : Install Composer] *********************************************
changed: [localhost]

TASK [composer : Install Composer Dependencies (Development)] ******************
skipping: [localhost]

TASK [composer : Install Composer Dependencies (Production)] *******************
changed: [localhost]

TASK [azuracast-db : Clear AzuraCast Cache] ************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "php /var/azuracast/www/util/cli.php cache:clear", "delta": "0:00:00.029869", "end": "2017-02-02 02:53:28.904410", "failed": true, "rc": 1, "start": "2017-02-02 02:53:28.874541", "stderr": "", "stdout": "Could not open input file: /var/azuracast/www/util/cli.php", "stdout_lines": ["Could not open input file: /var/azuracast/www/util/cli.php"], "warnings": []}

RUNNING HANDLER [azuracast-user : send user notification] **********************

RUNNING HANDLER [influxdb : restart influxdb] **********************************

RUNNING HANDLER [nginx : restart nginx] ****************************************

RUNNING HANDLER [php : restart php-fpm] ****************************************

RUNNING HANDLER [mariadb : restart mysql] **************************************

RUNNING HANDLER [mariadb : send mysql notification] ****************************
	to retry, use: --limit @/var/azuracast/www/util/ansible/deploy.retry

PLAY RECAP *********************************************************************
localhost                  : ok=69   changed=54   unreachable=0    failed=1   
```

Going to the address of the server in the web browser shows:

```
500

An exception occurred while executing 'SELECT s0_.setting_key AS setting_key_0, s0_.setting_value AS setting_value_1 FROM settings s0_ ORDER BY s0_.setting_key ASC': SQLSTATE[42S02]: Base table or view not found: 1146 Table 'azuracast.settings' doesn't exist
/var/azuracast/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php : 53
```

**Comment by teardroponfire** [comment]
I tried an installation on Ubuntu Desktop 16.04.1 LTS in VirtualBox and got an additional error that I didn't get with the VM. After after **TASK [azuracast-db : Clear AzuraCast Cache]**, I saw this:

```
TASK [azuracast-db : Install Initial Database] ***************************************
fata: [localhost]: FAILED => {"changed": false, "failed": true, "msg": "unable to find /home/kev/.my.cnf. Exception message: (1045, \"Access denied for user 'root'@'localhost' (using password: NO)\")"}
```

I was running the install script as root (by using `sudo -s`). It looks like the install script is looking for the .my.cnf file containing the root password in my non-root user home directory (/home/kev). I did not get this message when I ran the install on Ubuntu Server 16.04.1 LTS.
